### PR TITLE
added notification preferences crud

### PR DESCRIPTION
**Pull request by jleung40**
# Description

set up crud for notification preferences. temporary settings, to be changed as we get updates from design team on what settings should exist. 

## How has this been tested?

notification preferences tests

## Checklist

* [x]  I have self-reviewed my code for readability, maintainability, performance, and added comments/documentation where necessary.
* [x] New and existing tests pass locally with my changes.
* [x] I have followed the project's coding standards and best practices.

### User-visible changes
- Adds user-level notification preferences API at /api/v1/users/me/notification-preferences with GET, POST, PATCH, DELETE and POST /default
- Six toggleable notification categories (push_enabled, upcoming_trip, voting_reminders, finalized_decisions, trip_activity, deadline_reminders) default to true on creation
- Authentication and user isolation enforced; unauthenticated => 401, missing prefs => 404, duplicate create => 409
- Comprehensive end-to-end tests covering lifecycle, authorization, defaults, and boolean edge cases

### Changes by category

**Features**
- Controller: NotificationPreferencesController with handlers for Get, Create, Update, Delete, GetOrCreateDefault and request validation
- Service: NotificationPreferencesService (business logic for get/create/update/delete/get-or-create-default)
- Repository: NotificationPreferencesRepository interface and bun-backed implementation (Create, Find, Update partial, Delete, Upsert)
- Routing: mounted routes under API v1 (/users/me/notification-preferences)

**Infra**
- DB migration: adds notification_preferences table (user_id PK FK → users(id) ON DELETE CASCADE; six boolean flags default true; created_at/updated_at timestamps)
- Repository wiring: Repository struct now exposes NotificationPreferences repo
- Minor test mock import tweak (S3 presign client mock)

**Testing**
- New integration tests (notification_preferences_test.go) exercising full CRUD lifecycle, authorization, defaults, and edge cases

### Author contribution

| Category                    | Added | Removed | Net |
|----------------------------:|------:|-------:|----:|
| Controllers                 | 166   | 0     | 166 |
| Services                    | 95    | 0     | 95  |
| Repository (new file)       | 143   | 0     | 143 |
| Repository (repo.go)        | 40    | 38    | 2   |
| Router (routes file)        | 24    | 0     | 24  |
| Routers (init)             | 1     | 0     | 1   |
| Models                      | 35    | 1     | 34  |
| Migration                   | 19    | 0     | 19  |
| Tests                       | 467   | 0     | 467 |
| Misc (mocks)                | 1     | 1     | 0   |
| **Total**                   | **991** | **40** | **951** |

### CODE REVIEW

File: backend/internal/controllers/notification_preferences.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+// @Success      201 {object} models.NotificationPreferences
+// @Failure      400 {object} errs.APIError
+// @Failure      401 {object} errs.APIError
+// @Failure      409 {object} errs.APIError
+// @Failure      422 {object} errs.APIError
+// @Failure      500 {object} errs.APIError
+// @Router       /api/v1/users/me/notification-preferences [post]
+// @ID           createNotificationPreferences
+func (n *NotificationPreferencesController) CreatePreferences(c *fiber.Ctx) error {
+	userID, err := validators.ValidateID(c.Locals("userID").(string))
```
Comment (in-mai-space) [inline backend/internal/controllers/notification_preferences.go]: ditto the util here and rest of the file
